File: backend/src/main/java/com/arquitecsoft/gestion/domain/usuario/service/UsuarioService.java
package com.arquitecsoft.gestion.domain.usuario.service;

import com.arquitecsoft.gestion.domain.usuario.dto.UsuarioResponse;
import com.arquitecsoft.gestion.domain.usuario.repository.GcUsuarioRepository;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * Lectura de usuarios para selección (p. ej. propietario de empresa/persona — F-RP4/F-RP5).
 */
@Service
public class UsuarioService {

    private final GcUsuarioRepository usuarioRepository;

    public UsuarioService(GcUsuarioRepository usuarioRepository) {
        this.usuarioRepository = usuarioRepository;
    }

    /** Usuarios activos, ordenados por nombre, para selección como propietario/responsable. */
    @Transactional(readOnly = true)
    public List<UsuarioResponse> listarActivos() {
        return usuarioRepository.findAll(Sort.by("apellidos", "nombres").ascending()).stream()
                .filter(u -> u.getActivo() != null && u.getActivo() == 1)
                .map(UsuarioResponse::fromEntity)
                .toList();
    }
}

File: frontend/src/services/usuario.service.js
import api from './api';

const BASE = '/usuarios';

/**
 * Usuarios activos, para selección como propietario/responsable (F-RP4/F-RP5).
 * Sin try/catch: los errores burbujean a la vista (convención del repo).
 */
export const usuarioService = {
  async listar() {
    const { data } = await api.get(BASE);
    return data;
  },
};

export default usuarioService;
